index.js

```diff
@@ -17,4 +13,18 @@ module.exports = obj => {
       return { [entry[0]]: entry[1] };
     })
     .reduce((acc, curr) => Object.assign(acc, curr), {});
+}
+
+module.exports = obj => {
+  const promisifiedObject = promisifyObject(obj);
+
+  if (typeof obj === 'function') {
+    const promisiedFunction = promisify(obj);
+
+    Object.assign(promisiedFunction, promisifiedObject);
```

Comment:
This makes me feel dirty, how illegal is this? I can do `Object.entries` again, I suppose - is that better?